ir/irutils.cpp

```diff
@@ -272,4 +273,36 @@ std::vector<const Expression *> flattenListOrStructExpression(const Expression *
                       listLikeExpr->node_type_name());
 }
 
+template <typename Stmts>
+const IR::Node *inlineBlockImpl(const Transform &t, Stmts &&stmts) {
+    if (stmts.size() == 1) {
+        // it could also be a declaration, and it that case, we need to wrap it in a block anyway
+        if (auto *stmt = (*stmts.begin())->template to<IR::Statement>()) {
+            return stmt;
+        }
+    }
+    if (t.getParent<IR::BlockStatement>()) {
+        return new IR::IndexedVector<IR::StatOrDecl>(std::forward<Stmts>(stmts));
+    }
+    Util::SourceInfo srcInfo;
+    if (stmts.size() > 0) {  // no .empty in initializer_list!
```

Comment:
`std::empty`? I guess that won't work for the other two...

Reply:
It should work for anything that has .empty on c-style array and on initializer_list. But I could not decide if it is nicer or uglier than size() > 0.